core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala

```diff
@@ -1415,6 +1415,79 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with BeforeAndAfterE
       super.fetchBlockSync(host, port, execId, blockId)
     }
   }
+
+  def testGetOrElseUpdateForLargeBlock(storageLevel: StorageLevel) {
```

Comment:
shall we do them together in a follow-up PR? I think the test case in `DiskStoreSuite` is enough.